datafusion/core/src/physical_optimizer/enforce_distribution.rs

```diff
@@ -3302,6 +3278,12 @@ mod tests {
             "ParquetExec: file_groups={2 groups: [[x], [y]]}, projection=[a, b, c, d, e], output_ordering=[a@0 ASC]",
         ];
         assert_optimized!(expected, exec, true);
+        let expected = &[
```

Comment:
This seems like a regression -- the comments in the code say "the optimizer should not add a new sort exec" but then in this code the optimizer has added the new sort exec.

Maybe we need to update the test to add a `GlobalRequirements` node so that the sorts are not added?

the same comments apply to the other tests in this file